Problem: memiavl snapshot format is not optimal (#890)
crypto-org-chain/cronos: memiavl/README.md

* 27 Jan 2023:
  * Update metadata file format
  * Encode key length with 4 bytes instead of 2.

> * 21 Feb 2023:

Author: ```suggestion
* 24 Feb 2023:
```

  ```
  magic: uint32
  format: uint32
  version: uint32
  root node index: uint32
  ```

- `nodes`, array of fixed size(16+32bytes) nodes, the node format is like this:

  ```
  height   : uint32         // padded to 4bytes
  version  : uint32
  size     : uint32
  key_node : uint32
  hash     : [32]byte
  ```
  The node has fixed length, can be indexed directly. The nodes reference each other with the node index, nodes are written in post-order, so the root node is always placed at the end.

>   For branch node, the `key_node` field reference the minimal leaf node in the right branch for branch node, for leaf node, it's the leaf index instead, which can be used to find key and value in `keys` and `values` file.

Author: ```suggestion
  For branch node, the `key_node` field reference the smallest leaf node in the right branch, for leaf node, it's the leaf index, which can be used to find key and value in `keys` and `values` file.
```

>   The branch node don't need extra left/right children references, those are derived on the fly based on property of post-order traversal:

Author: ```suggestion
  The branch node don't need to reference left/right children explicitly, they can be derived from existing information and properties of post-order traversal:
```

  ```
  right child index = self index - 1
  left child index = key_node - 1
  ```

>   Some integers are using `uint32`, should be enough in forseeable future, but could be changed to `uint64` to be safer.

Author: ```suggestion
  The version/size/node indexes are encoded with `uint32`, should be enough in foreseeable future, but could be changed to `uint64` in the future.
```
